internal/provider/resource_vault_secrets_app.go

```diff
@@ -0,0 +1,169 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	clients "github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func NewVaultSecretsAppResource() *resourceVaultsecretsApp {
```

Comment:
nit: typically, this function returns a `resource.Resource`, as the struct `resourceVaultsecretsApp` should implement that interface

Reply:
Ahh. Good catch. Thanks